modular_skyrat/master_files/code/modules/clothing/towels.dm

```diff
@@ -0,0 +1,523 @@
+		return
+
+	if(flags_1 & HOLOGRAM_1) // Just in case there's ever holographic towels.
+		return
+
+	var/obj/item/stack/sheet/cloth/shreds = new (get_turf(src), TOWEL_CLOTH_AMOUNT)
+
+	if(!QDELETED(shreds)) //stacks merged
+		transfer_fingerprints_to(shreds)
+		shreds.add_fingerprint(user)
+
+	to_chat(user, span_notice("You tear [src] up into cloth."))
+	qdel(src)
+
+
+/obj/item/towel/pre_attack_secondary(atom/target, mob/living/user, params)
+	. = ..()
+
+	if(!istype(target, /obj/item/reagent_containers/cup/bucket))
+		return
+
+	if(!reagents.total_volume)
+		to_chat(user, span_warning("\The [src] is dry, you can't squeeze anything out!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	var/obj/item/reagent_containers/cup/bucket/target_bucket = target
+
+	if(target_bucket.reagents.total_volume >= target_bucket.reagents.maximum_volume)
+		to_chat(user, span_warning("[target] is full!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+	transfer_towel_reagents_to(target_bucket, reagents.total_volume, user, loss_factor = SQUEEZING_DISPERSAL_RATIO, make_used = TRUE) // If it didn't have enough space, oh well, you lost like 3/4th of what was in the towel anyway, there's just even more loss that way. Doesn't really matter.
+
+	to_chat(user, span_notice("You wring the liquid out of [src], transferring some of it to [target]."))
+
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+
+/obj/item/towel/AltClick(mob/user)
+	. = ..()
+
+	if(. == FALSE)
+		return
+
+	if(!(shape == TOWEL_FULL || shape == TOWEL_WAIST))
+		return FALSE
+
+	if(!ishuman(user))
+		return FALSE
+
+	var/mob/living/carbon/human/towel_user = user
+	var/worn = towel_user.wear_suit == src
+
+	change_towel_shape(user, shape == TOWEL_FULL ? TOWEL_WAIST : TOWEL_FULL, silent = worn)
+
+	// No need to display the different message if they're not wearing it.
+	if(!worn)
+		return
+
+	to_chat(user, span_notice(shape == TOWEL_FULL ? "You raise \the [src] over your [shape]." : "You lower \the [src] down to your [shape]."))
+
+
+/obj/item/towel/CtrlClick(mob/user)
+	. = ..()
+
+	if(. == FALSE)
+		return
+
+	if(!wet && shape == TOWEL_FOLDED) // You can't fold a wet towel, so you can't get a folded towel that's also wet. And you can't fold what's already folded, obviously.
+		to_chat(user, span_warning("You can't fold a towel that's already folded!"))
+		return
+
+	if(!ishuman(user) || !iscyborg(user))
+		if(iscyborg(user) && wet) // Cyborgs can't wring towels.
+			to_chat(user, span_warning("Folding a wet towel doesn't really make sense. You stop yourself before doing that."))
+			return
+
+		var/in_hands = TRUE
+
+		if(ishuman(user))
+			in_hands = user.get_active_held_item() == src || user.get_inactive_held_item() == src
+
+
+		if(!in_hands) // They need to be in your hands, unless you're a cyborg.
+			return
+
+		if(!wet)
+			change_towel_shape(user, TOWEL_FOLDED, silent = TRUE)
+			to_chat(user, span_notice("You fold [src] up neatly."))
+			return
+
+		// No cyborgs past this point.
```

Comment:
comment isn't quite accurate

Reply:
Now it will be, I messed up my first if() logic initially, whoops. But yeah, cyborgs with wet towels are caught prior to this point, and then not wet towels are caught here, so it's cyborg-proof past that.